catalog/core/catalog-core-downloadaction/src/test/java/org/codice/ddf/catalog/download/action/ResourceDownloadActionEndpointTest.java

```diff
@@ -0,0 +1,109 @@
+/**
+ * Copyright (c) Codice Foundation
+ * <p>
+ * This is free software: you can redistribute it and/or modify it under the terms of the GNU Lesser
+ * General Public License as published by the Free Software Foundation, either version 3 of the
+ * License, or any later version.
+ * <p>
+ * This program is distributed in the hope that it will be useful, but WITHOUT ANY WARRANTY; without
+ * even the implied warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+ * Lesser General Public License for more details. A copy of the GNU Lesser General Public License
+ * is distributed along with this program and can be found at
+ * <http://www.gnu.org/licenses/lgpl.html>.
+ */
+package org.codice.ddf.catalog.download.action;
+
+import static org.hamcrest.MatcherAssert.assertThat;
+import static org.hamcrest.Matchers.is;
+import static org.mockito.Mockito.doThrow;
+import static org.mockito.Mockito.when;
+
+import javax.management.MBeanException;
+import javax.ws.rs.core.Response;
+
+import org.codice.ddf.catalog.resource.download.DownloadToCacheException;
+import org.codice.ddf.catalog.resource.download.ResourceDownloadMBean;
+import org.junit.Test;
+import org.junit.runner.RunWith;
+import org.mockito.Mock;
+import org.mockito.runners.MockitoJUnitRunner;
+
+@RunWith(MockitoJUnitRunner.class)
+public class ResourceDownloadActionEndpointTest {
+
+    private static final String METACARD_ID = "57a4b894e13a455b8cccb87cec778b58";
+
+    private static final String SOURCE_ID = "ddf.distribution";
+
+    private static final String SUCCESS_MESSAGE_TEMPLATE = "Successfully started download of resource associated with metacard id %s from source %s";
+
+    @Mock
+    private ResourceDownloadMBean mockResourceDownloadMBeanProxy;
+
+    @Test(expected = DownloadToCacheException.class)
+    public void testDownloadToCacheMBeanProxyThrowsException() throws Exception {
+        // Setup
+        setupMockResourceDownloadMBeanWithException();
+        ResourceDownloadActionEndpoint resourceDownloadEndpoint = createResourceDownloadActionEndpoint();
+
+        // Perform Test
+        resourceDownloadEndpoint.downloadToCache(SOURCE_ID, METACARD_ID);
+    }
+
+    @Test(expected = DownloadToCacheException.class)
+    public void testDownloadToCacheNullSourceId() throws Exception {
+        // Setup
+        setupMockResourceDownloadMBeanWithException();
```

Comment:
We probably want to use `setupMockResourceDownloadMBean()` here, not the one that throws an exception.